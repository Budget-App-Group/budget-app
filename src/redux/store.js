import { createStore, applyMiddleware, combineReducers } from 'redux'
import promiseMiddleware from 'redux-promise-middleware'
import { composeWithDevTools } from 'redux-devtools-extension'

import userReducer from './userReducer'
import budgetReducer from './budgetReducer'

const rootReducers = combineReducers({
    user: userReducer,
    budget: budgetReducer
})

export default createStore(rootReducers, composeWithDevTools(applyMiddleware(promiseMiddleware)))