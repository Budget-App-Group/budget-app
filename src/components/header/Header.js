import React, { Component } from "react";
import { connect } from "react-redux";
import { checkUser, logout } from "../../redux/userReducer"
import { Link } from "react-router-dom";
import "./header.scss";

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      displayMenu: false,
    };
    this.logoutClicked = this.logoutClicked.bind(this)
  }

  logoutClicked() {
    this.props.logout()
  }
  

  showDropdownMenu = (event) => {
    event.preventDefault();
    this.setState({ displayMenu: true }, () => {
      document.addEventListener("click", this.hideDropdownMenu);
    });
  };

  hideDropdownMenu = () => {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener("click", this.hideDropdownMenu);
    });
  };

  render() {
    return (
      <div className="header-main">
        <div className="title">
          <h1>BudKid</h1>
        </div>
        <div
          className="dropdown"
          style={{ background: "white", width: "100px" }}
        >
          <div
            className="head-button"
            data-testid="menu-button"
            onClick={this.showDropdownMenu}
          >
            Menu
          </div>
          {this.state.displayMenu ? (
            <ul data-testid="dropdown">
              <li>
                <Link to="/admindashboard">Dashboard</Link>
              </li>
              <li>
                <Link to="/contactus">Contact Us</Link>
              </li>
              <li>
                <Link to="/chat">Chat</Link>
              </li>

              <li>
                {/* <a href="logut">Log Out</a> */}
                <div onClick={this.logoutClicked}>Logout</div>
              </li>
            </ul>
          ) : null}
        </div>
      </div>
    );
  }
}

const mapStateToProps = reduxState => {
  const { user } = reduxState.user
  return {
    user
  }
}

export default connect(mapStateToProps, { checkUser, logout })(Header)
