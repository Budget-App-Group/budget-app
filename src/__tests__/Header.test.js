import React from "react";
import { render, fireEvent } from "@testing-library/react";
// import axios from "axios";
import Header from "../components/header/Header";
import { MemoryRouter } from "react-router";
import { Provider } from "react-redux";
import store from "../redux/store";

it("does not show drop down when mounted", () => {
  const { queryByTestId } = render(
    <MemoryRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </MemoryRouter>
  );

  const dropdown = queryByTestId("dropdown");

  expect(dropdown).not.toBeTruthy();
});

it("shows dropdown when menu is clicked", () => {
  const { container, getByTestId } = render(
    <MemoryRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </MemoryRouter>
  );

  const headButton = getByTestId("menu-button");

  fireEvent.click(headButton);

  expect(container.textContent).toContain("Dashboard");
});
