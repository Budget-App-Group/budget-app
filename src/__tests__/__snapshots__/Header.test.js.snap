// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests Header component component header matches the snapshot 1`] = `
<div
  className="header-main"
>
  <div
    className="title"
    data-testid="title"
  >
    <h1>
      BudKid
    </h1>
  </div>
  <div
    className="dropdown"
    style={
      Object {
        "background": "white",
        "width": "100px",
      }
    }
  >
    <div
      className="head-button"
      data-testid="menu-button"
      onClick={[Function]}
    >
      Menu
    </div>
  </div>
</div>
`;
